### PR TITLE
Fix/modernizing-braces

### LINKED ISSUE #276: django-braces looks old
We wrote this back when 2.7 was in its heyday, Django CBVs had barely dropped. It was a different world. We need to bring this repo to modern Python.

### PR DESCRIPTION
**Pull request by kennethlove**
like any modern woman, I just want to have it all

Closes #276

This PR is a surface refactor of all of the `django-braces` code. This _should_ remove all of the Python 2 considerations and generally just make the code nicer.

I'll follow up with tickets for other updates/changes.

### CODE REVIEW

**Comment by kennethlove** [comment]
The "failed build" is because codecov was unavailable....

File: braces/views/_ajax.py [hunk 1]
```diff
@@ -56,30 +56,29 @@ def render_json_object_response(self, objects, **kwargs):
         return HttpResponse(json_data, content_type=self.get_content_type())
```
Comment (auvipy) [inline braces/views/_ajax.py:56]: why not  use django built in jsonresponse?

File: braces/forms.py [hunk 2]
```diff
@@ -9,6 +9,6 @@ class UserKwargModelFormMixin(object):
     """
 
     def __init__(self, *args, **kwargs):
-        self.user = kwargs.pop("user", None)  # Pop the user off the
-        # passed in kwargs.
+        """Remove the user from **kwargs and assign it on the object"""
+        self.user = kwargs.pop("user", None)
         super(UserKwargModelFormMixin, self).__init__(*args, **kwargs)
```
Comment (auvipy) [inline braces/forms.py:14]: this super call wont need explicit class call

File: braces/views/_access.py [hunk 3]
```diff
@@ -104,21 +105,19 @@ def dispatch(self, request, *args, **kwargs):
         if not request.user.is_authenticated:
             return self.handle_no_permission(request)
 
-        return super(LoginRequiredMixin, self).dispatch(
+        return super().dispatch(
             request, *args, **kwargs
         )
```
Comment (CleitonDeLima) [inline braces/views/_access.py:110]:
```suggestion
        return super().dispatch(request, *args, **kwargs)
```

File: braces/views/_access.py [hunk 4]
```diff
@@ -132,36 +131,33 @@ class SomeView(AnonymousRequiredMixin, ListView):
     def dispatch(self, request, *args, **kwargs):
         if request.user.is_authenticated:
             return HttpResponseRedirect(self.get_authenticated_redirect_url())
-        return super(AnonymousRequiredMixin, self).dispatch(
+        return super().dispatch(
             request, *args, **kwargs
         )
```
Comment (CleitonDeLima) [inline braces/views/_access.py:136]:
```suggestion
        return super().dispatch(request, *args, **kwargs)
```

File: braces/views/_ajax.py (hunk 1 above)
Comment (kennethlove) [reply braces/views/_ajax.py:56]: I don't think it existed at the time when we wrote this mixin. Good call, though.

Comment (kennethlove) [reply braces/views/_ajax.py:56]: I think I'll leave that refactor for later on, though. Odds are good we should just drop the mixin.

**Pull request merged**